### PR TITLE
MAINT: Update byte_bounds import

### PR DESCRIPTION
**Pull request by mtsokol**
This PR addresses changes introduced in https://github.com/numpy/numpy/pull/24540 and https://github.com/numpy/numpy/pull/24376 relevant to NEP 52.

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
LGTM.

In the unlikely event that someone is going to complain about the `array_utils` name and we end up changing it in numpy's `main` branch, the code can be updated. This should be safe to merge now.

File: joblib/_memmapping_reducer.py [hunk 1]
```diff
@@ -266,8 +266,12 @@ def _reduce_memmap_backed(a, m):
     # offset that comes from the striding differences between a and m
     util.debug('[MEMMAP REDUCE] reducing a memmap-backed array '
                '(shape, {}, pid: {})'.format(a.shape, os.getpid()))
-    a_start, a_end = np.byte_bounds(a)
-    m_start = np.byte_bounds(m)[0]
+    try:
+        from numpy.lib.array_utils import byte_bounds
+    except (ModuleNotFoundError, ImportError):
+        from numpy import byte_bounds
```
Comment (ogrisel) [inline joblib/_memmapping_reducer.py:272]:
```suggestion
        # Backward-compat for numpy < 2.0
        from numpy import byte_bounds
```

**Review by ogrisel — APPROVED** [review]

**Comment by ogrisel** [comment]
Thanks, could you please just update the changelog to make it explicit that the next joblib release targets numpy 2.0 compat?

**Comment by mtsokol** [comment]
> Thanks, could you please just update the changelog to make it explicit that the next joblib release targets numpy 2.0 compat?

Sure! I added it.

**Review by ogrisel — APPROVED** [review]

**Comment by ogrisel** [comment]
The CI failure is weird but unrelated to this PR.

**Pull request merged**